### PR TITLE
MAINT: special: remove one `libnpymath` dependency; more `NPY_` -> C99

### PR DESCRIPTION
**Pull request by rgommers**
Replacements:
- `NPY_NAN` -> `NAN`
- `NPY_INFINITY` -> `INFINITY`
- `NPY_PI` -> `M_PI`
- `NPY_E` -> `M_E`
- `npy_isnan` -> `isnan`
- `npy_isinf` -> `isinf`
- `npy_isfinite` -> `isfinite`
- `npy_copysign` -> `copysign`
- `npy_signbit` -> `signbit`

Also cleans up a lot of Cython imports, and removes one `libnpymath` inclusion. That removal reduces binary size and works towards not depending on `libnpymath` at all - see https://github.com/numpy/numpy/issues/20880.

Things brings us down to 13 places where we still include `numpy/npy_math.h`.

### CODE REVIEW

File: scipy/special/cdf_wrappers.c [hunk 1]
```diff
@@ -29,25 +29,25 @@
...
 
 #define CDFLIB_CALL2(func, name, a, b, result, return_bound)            \
-    if (npy_isnan(p) || npy_isnan(q) || npy_isnan(a) || npy_isnan(b) || \
-        npy_isnan(bound)) {                                             \
-        return NPY_NAN;                                                 \
+    if (isnan(p) || isnan(q) || isnan(a) || isnan(b) || \
+        isnan(bound)) {                                             \
+        return NAN;                                                 \
     }                                                                   \
     func(&which, &p, &q, &a, &b, &status, &bound);                      \
```
Comment (h-vetinari) [inline scipy/special/cdf_wrappers.c:36]: Would be nice to keep the formatting of the macros intact.
```suggestion
    if (isnan(p) || isnan(q) || isnan(a) || isnan(b) || isnan(bound)) { \
        return NAN;                                                     \
    }                                                                   \
    func(&which, &p, &q, &a, &b, &status, &bound);                      \
```

(same below)

**Review by h-vetinari — COMMENTED** [review]
Just a quick pass, but looks like a great clean-up!

**Comment by rgommers** [comment]
The one failure (`TestSchur.test_sort`) is gh-14517, unrelated to this PR. I'll cancel CI for the second commit (please squash-merge that)

File: scipy/special/cdf_wrappers.c [hunk 2]
```diff
@@ -29,25 +29,24 @@
...
     return get_result(name, status, bound, result, return_bound)
 
 #define CDFLIB_CALL3(func, name, a, b, c, result, return_bound)         \
-    if (npy_isnan(p) || npy_isnan(q) || npy_isnan(a) || npy_isnan(b) || \
-        npy_isnan(c) || npy_isnan(bound)) {                             \
-        return NPY_NAN;                                                 \
+    if (isnan(p) || isnan(q) || isnan(a) || isnan(b) || \
+        isnan(c) || isnan(bound)) {                             \
+        return NAN;                                                 \
     }                                                                   \
```
Comment (h-vetinari) [inline scipy/special/cdf_wrappers.c:42]:
```suggestion
    if (isnan(p) || isnan(q) || isnan(a) || isnan(b) ||                 \
        isnan(c) || isnan(bound)) {                                     \
        return NAN;                                                     \
    }                                                                   \
```

File: scipy/special/cdf_wrappers.c [hunk 3]
```diff
@@ -29,25 +29,24 @@
...
     return get_result(name, status, bound, result, return_bound)
 
 #define CDFLIB_CALL4(func, name, a, b, c, d, result, return_bound)      \
-    if (npy_isnan(p) || npy_isnan(q) || npy_isnan(a) || npy_isnan(b) || \
-        npy_isnan(c) || npy_isnan(d) || npy_isnan(bound)) {             \
-        return NPY_NAN;                                                 \
+    if (isnan(p) || isnan(q) || isnan(a) || isnan(b) || \
+        isnan(c) || isnan(d) || isnan(bound)) {             \
+        return NAN;                                                 \
     }                                                                   \
```
Comment (h-vetinari) [inline scipy/special/cdf_wrappers.c:50]:
```suggestion
    if (isnan(p) || isnan(q) || isnan(a) || isnan(b) ||                 \
        isnan(c) || isnan(d) || isnan(bound)) {                         \
        return NAN;                                                     \
    }                                                                   \
```

**Review by h-vetinari — COMMENTED** [review]
> (same below)

There were two more instances...

**Comment by ev-br** [comment]
LGTM. Squash-merged as requested to keep the ball rolling, thanks Ralf, all.

**Pull request merged**